frontend/packages/kubevirt-plugin/src/components/vms/menu-actions.tsx

```diff
@@ -136,12 +148,58 @@ const menuActionCancelMigration = (
   };
 };
 
+const menuActionClone = (kindObj: K8sKind, vm: VMKind, { vmStatus }: ActionArgs): KebabOption => {
+  return {
+    hidden: isVMImporting(vmStatus),
+    label: 'Clone Virtual Machine',
+    callback: () => {
+      const namespace = getNamespace(vm);
+      const resources = [
+        getResource(NamespaceModel, { prop: 'namespaces' }),
+        getResource(VirtualMachineModel, { prop: 'virtualMachines' }),
+        getResource(PersistentVolumeClaimModel, { namespace, prop: 'persistentVolumeClaims' }),
+        getResource(DataVolumeModel, { namespace, prop: 'dataVolumes' }),
+      ];
+
+      const resourcesToProps = (props) => ({
+        vm,
+        ...resources
+          .map((r) => r.prop)
+          .reduce((mappedResources, propName) => {
+            mappedResources[propName] = getLoadedData(props[propName]);
```

Comment:
I think we should handle the case when the user has not enough privileges to list these resources.